### PR TITLE
The release config can now specify additional in-release json

### PR DESCRIPTION
**Pull request by timbod7**
files to be merged into the config interpolation context

### CODE REVIEW

File: adl/release.adl [hunk 1]
```diff
@@ -11,6 +11,10 @@ struct ReleaseConfig
     String prestartCommand;
     String startCommand;
     String stopCommand;
+
+    // Files inside the release to be injected into
+    // the template at release time.
```
Comment (jsuresh) [inline adl/release.adl:16]: Is this an absolute path from the release root? Worth updating comment with an example path

I find the comment in the ADL useful when trying to setup and get c2 working :)

File: src/Util.hs [hunk 2]
```diff
@@ -155,20 +157,30 @@ checkReleaseExists release = do
...
 
-  return (JS.Object (HM.fromList ([(T.pack file,jv) | (file,jv) <- values]) ))
+  return (HM.fromList ([(T.pack file,jv) | (file,jv) <- values]) )
+
+-- Load template context from the release package
+loadMergedReleaseContext :: FilePath -> ReleaseConfig -> IO (HM.HashMap T.Text JS.Value)
+loadMergedReleaseContext dir rcfg = do
+
+  values <- for (M.toList (SM.toMap (rc_configSources rcfg))) $ \(name,src) -> do
+    let filepath = dir </> T.unpack src
```
Comment (jsuresh) [inline src/Util.hs:177]: Guessing this implies no leading '/' in the config var? Does this mean (if I wanted) I could also go ../ (not that I'd want to, nor do I think it's a security issue - just trying to understand the code)

**Review by jsuresh — APPROVED** [review]
LGTM

File: src/Util.hs (hunk 2 above)
Comment (timbod7) [reply src/Util.hs:177]: A leading / in the path is not intended, but will work, as per docs for [`</>`](
https://hackage.haskell.org/package/filepath-1.4.2.1/docs/System-FilePath-Posix.html#v:-60--47--62-
)
`..` would also work.

File: adl/release.adl (hunk 1 above)
Comment (timbod7) [reply adl/release.adl]: I've added a comment

**Pull request merged**